Sources/PackageModel/ToolsVersionSpecificationGeneration.swift

```diff
@@ -0,0 +1,47 @@
+// PackageModel/ToolsVersionSpecificationGeneration.swift
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2020 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+// -----------------------------------------------------------------------------
+///
+/// This file contains an extension to `ToolsVersion` that implements the generation of a Swift tools version specification from a `ToolsVersion` instance.
+///
+// -----------------------------------------------------------------------------
+
+extension ToolsVersion {
+    // TODO: Add options for whitespace styles.
+    /// Returns a Swift tools version specification specifying the version to the given precision.
+    /// - Parameter resolution: The precision to which the version specifier follows the version.
+    /// - Returns: A  Swift tools version specification specifying the version to the given precision.
+    public func specification(resolution: SpecifierResolution = .automatic) -> String {
```

Comment:
The parameter should be named better than `resolution`, but I can't think of a better one.

Reply:
What do you think of `rounding`, `rounded`, or `significantThrough`?

I think making the enum more specific to the patch version works too, but it will need more changes when/if `ToolsVersion` starts to carry version metadata.